### PR TITLE
ICU-22283 Add additional ERoundingMode variants

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22283
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by richgillam — COMMENTED** [review]
Most of this looks fine, but I couldn't follow what was going on the unit tests.

There were a couple of things: First, I didn't understand what the third column was, the one that sometimes had a "K" or some other letter (or sequence of letters) in it.  Second, it seems like you're modifying a lot of existing tests, not just adding new ones to cover the new rounding modes, which suggests the rounding-mode change has some kind of side effect (or that I'm just misinterpreting what you're doing).  Could you elucidate?

**Comment by sffc** [comment]
I change numberformattestspecification.txt to run on Java 11 instead of Java 8 so that I could reproduce test failures, which resulted in my having to update more of the tests in that file whose behavior changed between 8 and 11. This change is in the last file in the PR.

**Review by richgillam — APPROVED** [review]

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatDataDrivenTest.java
Comment (markusicu) [inline icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatDataDrivenTest.java:719]: This test runs only on one version, right? -->
```suggestion
        // #13373: Since not all JDK implementations are the same, test only on a JDK
        // with known behavior. The JDK version should be occasionally updated.
```

On other JDKs/versions, does it fail? Or does `assumeTrue()` somehow exit this test function?

Comment (sffc) [reply icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatDataDrivenTest.java]: I think that is how assumeTrue works; the test shows as skipped in JUnit out

**Review by markusicu — COMMENTED** [review]
lgtm tnx
(ignoring the numberformattestspecification.txt changes...)

**Comment by sffc** [comment]
@FrankYFTang lgty?

**Review by FrankYFTang — DISMISSED** [review]
LGTM
Thanks for working on this.

**Comment by FrankYFTang** [comment]
Please squash the branch

**Comment by sffc** [comment]
I think this can be merged once someone approves the squash

**Review by richgillam — APPROVED** [review]
LGTM

**Review by FrankYFTang — APPROVED** [review]

**Pull request merged**
